docs/best_practices/naming.rst

```diff
@@ -2,52 +2,70 @@ Naming
-choose another name. Keep in mind that meta-data should not be stored solely in the name of objects. It is OK to name
-an object something like “ElectricalSeries_large_array” however the name alone is not sufficient documentation. In this
-case, the source of the signal will be clear from the device of the rows from the linked electrodes table region, and
-you should also include any important distinguishing information in the description field of the object. Make an effort
-to make meta-data as explicit as possible. Good names help users but do not help applications parse your file.
+Metadata Descriptions
+~~~~~~~~~~~~~~~~~~~~~
+
+The ``name`` of an object is for identifying that object within the file; it is not for storing metadata. Instead, the
+``description`` field for most objects can be used to include any important distinguishing information. When writing
+metadata, make every effort to be as explicit and self-contained as possible in the explaining relevant details.
+
+It is OK to name an object something like ``ElectricalSeriesDuringSomeEvent``, however the name alone is not sufficient
+documentation of the condition ``DuringSomeEvent``. In this case, the source of the signal will be clear from the
+:nwb-schema:ref:`device` linkage in the rows of the passed :nwb-schema:ref:`sec-sec-dynamictableregion` subsetting
+the full :nwb-schema:ref:`ElectrodeTable <groups-general-extracellular-ephys-electrodes>`, so you would not need to
+add any explicit metadata explaining these details.
+
+Slash-like characters ``'/'``, ``'\'``, or ``'|'``  are not not allowed in the ``name`` of an object. This can be
+confusing to systems that parse HDF5 files (which NWB uses, see the
+:nwb-overview:`NWB FAQ <faq_details/why_hdf5.html#why-hdf5>` for more details) because similar protocols are used to
+specify the location of ``groups`` within the file.
+
+For mathematical expressions, instead of including the special character in the name, please use an English equivalent
+instead. *E.g.*, instead of "Df/f" use "DfOverF".
 
-’/’ is not allowed in names. When creating a custom name, using the forward slash (/) is not allowed, as this can
-confuse h5py and lead to the creation of an additional group. Instead of including a forward slash in the name, please
-use “Over” like in DfOverF.
 
 
 Processing Modules
 ------------------
 
 
+.. _best_practice_processing_module_names:
+
 Indicate Modality
 ~~~~~~~~~~~~~~~~~
 
-Give preference to default processing module names. These names mirror the extension module names: “ecephys”,
-“icephys”, “behavior”, “ophys”, “misc”.
-
-We encourage the use of these defaults, but there may be some cases when deviating from this pattern is appropriate.
+Give preference to default :nwb-schema:ref:`sec-processingmodule` names. These names mirror the most common modalities:
+``ecephys``, ``icephys``, ``behavior``, ``ophys``, ``misc``.
```

Comment:
I prefer the above text, which explains that these abbreviations are not arbitrary, but defined in the NWB schema.